samples/nrf_desktop/src/hw_interface/battery_discrete.c

```diff
@@ -77,6 +77,10 @@ static bool sampling;
 
 static void error_check_handler(struct k_work *work)
 {
+	if (unlikely(battery_state == BATTERY_STATE_ERROR)) {
```

Comment:
This is not needed - it is enought to cancel the work (as long as we will do it from synchronous place - i.e. system workqueue context).

Reply:
ok